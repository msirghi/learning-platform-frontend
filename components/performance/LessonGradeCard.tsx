import { Avatar, Button, IconButton } from '@material-ui/core';
import React, { useState } from 'react';
import styles from '../../styles/modules/Performance.module.scss';
import ReactCardFlip from 'react-card-flip';
import UndoIcon from '@material-ui/icons/Undo';
import { Grade, Mark } from '../../common/types';
import dateFormat from 'dateformat';
import GradeTestService from '../services/GradeTestService';
import { GradeType } from '../../common/enums';
import ColorService from '../services/ColorService';
import { useWindowSize } from '../common/hooks/useWindowResize';
import { useTranslation } from '../../i18n';

type Props = {
  grade: Grade;
};

type CellProps = {
  mark?: Mark;
  type: GradeType;
};

const Cell: React.FC<CellProps> = ({ mark, type }) => {
  return (
    <div className={styles.cell}>
      <span className={styles.title}>
        {GradeTestService.getRelevantTranslation(mark ? mark.test : type)}
      </span>
      <span className={styles.mark}>{mark ? mark.mark : 'n/a'}</span>
    </div>
  );
};

export const LessonGradeCard: React.FC<Props> = ({ grade }) => {
  const { course, marks, maxPresence, presence, teacher, year } = grade;
  const [flipped, setFlipped] = useState(false);
  const [width] = useWindowSize();
  const { t } = useTranslation();

  const handleFlipChange = () => setFlipped(!flipped);

  return (
    <ReactCardFlip isFlipped={flipped} flipDirection='vertical'>
      <div className={styles.lessonGradeCardWrapper}>
        <div>
          <Avatar
            className={styles.avatar}
            style={{ backgroundColor: ColorService.getAvatarColorByText(course) }}
          >
            {grade.course.substring(0, 2)}
          </Avatar>
        </div>
        <div className={styles.lessonInfo}>
          <div className={styles.year}>{year}</div>
          <div className={styles.name}>{course}</div>
          <div className={styles.teacher}>{teacher}</div>
        </div>

        <div className={styles.gradeWrapper}>
          <div className={styles.mark}>{marks[0].mark}</div>
          <div className={styles.name}>
            {GradeTestService.getRelevantTranslation(marks[0].test)}
          </div>
          <div className={styles.date}>{dateFormat(marks[0].updatedAt, 'mediumDate')}</div>
        </div>

        <div className={styles.buttonContainer}>
          <Button
            fullWidth
            variant={width < 768 ? 'outlined' : 'text'}
            color='primary'
            onClick={() => handleFlipChange()}
          >
            {t('performance:viewAll')}
          </Button>
        </div>
      </div>

      <div className={styles.marksContainer}>
        <div className={styles.decor}></div>

        <div className={styles.cellRow}>
          <Cell
            mark={marks.find((m) => m.test === GradeType.ATTESTATION_1)}
            type={GradeType.ATTESTATION_1}
          />
          <Cell
            mark={marks.find((m) => m.test === GradeType.ATTESTATION_2)}
            type={GradeType.ATTESTATION_2}
          />
        </div>

        <div className={styles.cellRow}>
          <Cell mark={marks.find((m) => m.test === GradeType.EXAM)} type={GradeType.EXAM} />
          <Cell
            mark={marks.find((m) => m.test === GradeType.INDIVIDUAL_WORK)}
            type={GradeType.INDIVIDUAL_WORK}
          />
        </div>

        <div className={styles.cellRow}>
          <div className={styles.cell}>
            <span className={styles.title}>{t('performance:presence')}</span>
            <span className={styles.mark}>
              {presence} / {maxPresence}
            </span>
          </div>
          <Cell mark={marks.find((m) => m.test === GradeType.FINAL)} type={GradeType.FINAL} />
        </div>

        <div className={styles.undoButton}>
          <IconButton onClick={handleFlipChange}>
            <UndoIcon color='primary' />
          </IconButton>
        </div>
      </div>
    </ReactCardFlip>
  );
};
