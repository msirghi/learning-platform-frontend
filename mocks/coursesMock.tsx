import { Course } from '../common/types';

export const coursesMock: Course[] = [
  {
    id: '1',
    name: 'Course 1',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Libero non illum est fuga beatae ab, assumenda reiciendis nesciunt numquam similique!`,
    coverImg:
      'https://upload.wikimedia.org/wikipedia/commons/thumb/b/b6/Image_created_with_a_mobile_phone.png/1200px-Image_created_with_a_mobile_phone.png',
    teacherImg:
      'https://dyl80ryjxr1ke.cloudfront.net/external_assets/hero_examples/hair_beach_v1785392215/original.jpeg'
  },
  {
    id: '2',
    name: 'Course 2',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    coverImg: 'https://www.w3schools.com/w3css/img_lights.jpg',
    teacherImg:
      'https://images.pexels.com/photos/614810/pexels-photo-614810.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500'
  },
  {
    id: '3',
    name: 'Course 3',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    coverImg:
      'https://helpx.adobe.com/content/dam/help/en/stock/how-to/visual-reverse-image-search/jcr_content/main-pars/image/visual-reverse-image-search-v2_intro.jpg',
    teacherImg:
      'https://i2-prod.mirror.co.uk/incoming/article14334083.ece/ALTERNATES/s1200c/3_Beautiful-girl-with-a-gentle-smile.jpg'
  },
  {
    id: '4',
    name: 'Course 4',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    coverImg:
      'https://www.metoffice.gov.uk/binaries/content/gallery/metofficegovuk/hero-images/advice/maps-satellite-images/satellite-image-of-globe.jpg',
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '5',
    name: 'Course 5',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '6',
    name: 'Course 6',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '7',
    name: 'Course 7',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '8',
    name: 'Course 8',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '9',
    name: 'Course 9',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '10',
    name: 'Course 10',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '11',
    name: 'Course 11',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '12',
    name: 'Course 12',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '13',
    name: 'Course 13',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  }
];
