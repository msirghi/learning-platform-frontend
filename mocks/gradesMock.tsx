import { GradeType } from '../common/enums';
import { Grade } from '../common/types';

export const GradesMock: Grade[] = [
  {
    id: '1',
    course: 'Mathematics',
    marks: [
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_1,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_2,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.EXAM,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.INDIVIDUAL_WORK,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.INDIVIDUAL_WORK,
        updatedAt: new Date()
      }
    ],
    teacher: 'Teacher 1',
    year: 'First year',
    maxPresence: 15,
    presence: 3
  },
  {
    id: '2',
    course: 'Computer Science Computer Science',
    marks: [
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_1,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_2,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.EXAM,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.INDIVIDUAL_WORK,
        updatedAt: new Date()
      }
    ],
    teacher: 'Teacher 2',
    year: 'First year',
    maxPresence: 15,
    presence: 7
  },
  {
    id: '3',
    course: 'Biology',
    marks: [
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_1,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_2,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.EXAM,
        updatedAt: new Date()
      }
    ],
    teacher: 'Teacher 3',
    year: 'First year',
    maxPresence: 20,
    presence: 15
  },
  {
    id: '4',
    course: 'Physics',
    marks: [
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_1,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_2,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.EXAM,
        updatedAt: new Date()
      }
    ],
    teacher: 'Teacher 4',
    year: 'First year',
    maxPresence: 20,
    presence: 19
  }
];
