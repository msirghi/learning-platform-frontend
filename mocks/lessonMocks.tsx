import { LessonStatus } from '../common/enums';
import { Lesson } from '../common/types';

export const lessonMock: Lesson[] = [
  {
    id: '22',
    status: LessonStatus.ENDED,
    topic: 'Topic 1',
    title: `Lorem ipsum dolor sit amet.`,
    time: '8:00',
    start: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      10,
      0,
      0,
      0
    ),
    end: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      10,
      30,
      0,
      0
    )
  },
  {
    id: '2',
    status: LessonStatus.ONGOING,
    topic: `Lorem ipsum dolor sit amet.`,
    title: 'Lesson 2',
    time: '9:30',
    start: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      13,
      0,
      0,
      0
    ),
    end: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      14,
      0,
      0,
      0
    )
  },
  {
    id: '3',
    status: LessonStatus.NOT_STARTED,
    topic: 'Topic 3',
    title: 'Lesson 3',
    time: '11:00',
    start: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      14,
      30,
      0,
      0
    ),
    end: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      15,
      30,
      0,
      0
    )
  },
  {
    id: '4',
    status: LessonStatus.NOT_STARTED,
    topic: 'Topic 4',
    title: 'Lesson 4',
    time: '13:00',
    start: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      16,
      0,
      0,
      0
    ),
    end: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      17,
      30,
      0,
      0
    )
  }
];
