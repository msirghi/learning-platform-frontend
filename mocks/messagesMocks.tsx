import { ChatMessage } from '../common/types';

export const messagesMock: ChatMessage[] = [
  {
    id: 1,
    author: 'Mihail',
    date: new Date(),
    message: `Lorem ipsum dolor sit amet.`
  },
  {
    id: 2,
    author: 'Nicolae',
    date: new Date(),
    message: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Praesentium, sapiente!`
  },
  {
    id: 3,
    author: 'Mihail 2',
    date: new Date(),
    message: `Lorem ipsum, dolor sit amet consectetur adipisicing.`
  },
  {
    id: 4,
    author: 'Mihail 4',
    date: new Date(),
    message: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusamus, repudiandae? Iusto labore dolores obcaecati maxime. Quaerat praesentium doloribus expedita voluptatum!`
  },
  {
    id: 5,
    author: 'Mihail 4',
    date: new Date(),
    message: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusamus, repudiandae? Iusto labore dolores obcaecati maxime. Quaerat praesentium doloribus expedita voluptatum!`
  },
  {
    id: 6,
    author: 'Mihail 4',
    date: new Date(),
    message: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusamus, repudiandae? Iusto labore dolores obcaecati maxime. Quaerat praesentium doloribus expedita voluptatum!`
  }
];
