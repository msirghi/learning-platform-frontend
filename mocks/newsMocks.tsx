import { News } from "../common/types";

export const newsMock: News[] = [
  {
    id: "1",
    date: new Date(),
    description: `Lorem ipsum dolor sit, amet consectetur adipisicing elit. Dignissimos nihil, commodi alias at debitis in.`,
    title: "News 1",
  },
  {
    id: "2",
    date: new Date(),
    description: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Officiis numquam nobis, nihil quis error repudiandae.`,
    title: "News 2",
  },
];
