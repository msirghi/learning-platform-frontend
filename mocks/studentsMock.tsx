import { Student } from '../common/types';

export const studentsMock: Student[] = [
  {
    id: 1,
    firstName: 'First',
    lastName: 'Last'
  },
  {
    id: 2,
    firstName: 'Nicolae',
    lastName: 'Budulae'
  },
  {
    id: 3,
    firstName: 'Budulae',
    lastName: 'Nicolae'
  },
  {
    id: 4,
    firstName: 'Aron',
    lastName: 'Pumnul'
  },
  {
    id: 5,
    firstName: 'First',
    lastName: 'Last'
  },
  {
    id: 6,
    firstName: 'First',
    lastName: 'Last'
  },
  {
    id: 7,
    firstName: 'First',
    lastName: 'Last'
  }
];
