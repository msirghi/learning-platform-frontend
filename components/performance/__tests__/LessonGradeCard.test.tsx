import { Button } from '@material-ui/core';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { GradesMock } from '../../../mocks/gradesMock';
import { LessonGradeCard } from '../LessonGradeCard';

describe('LessonGradeCard component', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<LessonGradeCard grade={GradesMock[1]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render flipped card on button click', () => {
    const wrapper = shallow(<LessonGradeCard grade={GradesMock[1]} />);
    const button = wrapper.find(Button);
    expect(button).toHaveLength(1);

    button.simulate('click');
    const flipped = wrapper.find('.marksContainer');
    expect(flipped).toHaveLength(1);
  });
});
