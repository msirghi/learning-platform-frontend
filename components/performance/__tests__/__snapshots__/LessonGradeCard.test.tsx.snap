// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LessonGradeCard component should match the snapshot 1`] = `
<ReactCardFlip
  cardStyles={
    Object {
      "back": Object {},
      "front": Object {},
    }
  }
  cardZIndex="auto"
  containerStyle={Object {}}
  flipDirection="vertical"
  flipSpeedBackToFront={0.6}
  flipSpeedFrontToBack={0.6}
  infinite={false}
  isFlipped={false}
>
  <div
    className="lessonGradeCardWrapper"
  >
    <div>
      <WithStyles(ForwardRef(Avatar))
        className="avatar"
        style={
          Object {
            "backgroundColor": "#f55723",
          }
        }
      >
        Co
      </WithStyles(ForwardRef(Avatar))>
    </div>
    <div
      className="lessonInfo"
    >
      <div
        className="year"
      >
        First year
      </div>
      <div
        className="name"
      >
        Computer Science Computer Science
      </div>
      <div
        className="teacher"
      >
        Teacher 2
      </div>
    </div>
    <div
      className="gradeWrapper"
    >
      <div
        className="mark"
      >
        9.12
      </div>
      <div
        className="name"
      />
      <div
        className="date"
      >
        Jun 15, 2015
      </div>
    </div>
    <div
      className="buttonContainer"
    >
      <WithStyles(ForwardRef(Button))
        color="primary"
        fullWidth={true}
        onClick={[Function]}
        variant="outlined"
      >
        performance:viewAll
      </WithStyles(ForwardRef(Button))>
    </div>
  </div>
  <div
    className="marksContainer"
  >
    <div
      className="decor"
    />
    <div
      className="cellRow"
    >
      <Cell
        mark={
          Object {
            "mark": "9.12",
            "test": "ATTESTATION_1",
            "updatedAt": 2015-06-14T22:12:05.275Z,
          }
        }
        type="ATTESTATION_1"
      />
      <Cell
        mark={
          Object {
            "mark": "9.12",
            "test": "ATTESTATION_2",
            "updatedAt": 2015-06-14T22:12:05.275Z,
          }
        }
        type="ATTESTATION_2"
      />
    </div>
    <div
      className="cellRow"
    >
      <Cell
        mark={
          Object {
            "mark": "9.12",
            "test": "EXAM",
            "updatedAt": 2015-06-14T22:12:05.275Z,
          }
        }
        type="EXAM"
      />
      <Cell
        mark={
          Object {
            "mark": "9.12",
            "test": "INDIVIDUAL_WORK",
            "updatedAt": 2015-06-14T22:12:05.275Z,
          }
        }
        type="INDIVIDUAL_WORK"
      />
    </div>
    <div
      className="cellRow"
    >
      <div
        className="cell"
      >
        <span
          className="title"
        >
          performance:presence
        </span>
        <span
          className="mark"
        >
          7
           / 
          15
        </span>
      </div>
      <Cell
        type="FINAL"
      />
    </div>
    <div
      className="undoButton"
    >
      <WithStyles(ForwardRef(IconButton))
        onClick={[Function]}
      >
        <Memo([object Object])
          color="primary"
        />
      </WithStyles(ForwardRef(IconButton))>
    </div>
  </div>
</ReactCardFlip>
`;
